### PR TITLE
Fix otaclient failed to recover from a RECOVERABLE ota failure

### PR DESCRIPTION
**Pull request by Bodong-Yang**
## Introduction
This PR fix the issue that prevent otaclient recovering from a RECOVERABLE ota failure(like network errors). The cause of this issue is the `OTAUpdateFSM` doesn't  take otaclient failure and ota service failure into account, make otaclient/ota_service keep waiting even ota_service/otaclient failed.

## Changes
1. update `OTAUpdateFSM`, implement `on_<otaclient/otaservice>_failed`methods to allow interrupting fsm on failure,
2. `OTAUpdateFSM` wait methods now support return a bool to indicate the waited event is successful or not,
3. updates `ota_client_stub` and `ota_client` according to `OTAUpdateFSM` changes,
4. define `E_OTAPROXY_FAILED_TO_START` OTAError type, ota_proxy lauching failed now will be reported by the otaclient.

## Tests

- [x] recovered from RECOVERABLE failure at pre_update/METADATA stage
- [x] recovered from RECOVERABLE failure at in_update/REGULAR stage

### CODE REVIEW

File: app/errors.py [hunk 1]
```diff
@@ -22,6 +22,7 @@ class OTAModules(Enum):
     StandbySlotCreater = 2
     Downloader = 3
     API = 4
+    OTASERVICE = 5
```
Comment (obi-t4) [inline app/errors.py:25]: Enum entry should be UPPER_SNAKE.
https://kamatimaru.hatenablog.com/entry/2021/10/20/230128

**Review by obi-t4 — APPROVED** [review]
LGTM :+1:

**Review by obi-t4 — COMMENTED** [review]
I left a comment but this is a minor issue.
I'd like to fix the naming rule:
OTAPROXY or OTA_PROXY, 
OTAClient or OtaClient
etc.

**Comment by Bodong-Yang** [comment]
> I left a comment but this is a minor issue. I'd like to fix the naming rule on another PR like:
> 
> ```
> OTAPROXY or OTA_PROXY, 
> OTAClient or OtaClient
> ```
> 
> etc.

Sure!

**Pull request merged**